### PR TITLE
Learning rate

### PR DESCRIPTION
**Pull request by ElArkk**
This PR makes the learning rate of the model a tuneable parameter, passed to optuna.

### CODE REVIEW

File: jax_unirep/evotuning.py [hunk 1]
```diff
@@ -328,6 +337,12 @@ def evotune(
         This controls how many epochs to have Optuna test.
         See source code for default configuration,
         at the definition of ``n_epochs_kwargs``.
+    :param learning_rate_config: A dictionary of kwargs
+        to ``trial.suggest_loguniform``,
+        which are: ``name``, ``low``, ``high``.
+        This controls how many epochs to have Optuna test.
+        See source code for default configuration,
+        at the definition of ``n_epochs_kwargs``.
```
Comment (ericmjl) [inline jax_unirep/evotuning.py]:
```suggestion
        Same purpose as `n_epochs_kwargs`.
```

File: jax_unirep/evotuning.py [hunk 2]
```diff
@@ -265,10 +265,19 @@ def objective(
         "high": len(sequences) * 3,
         "q": 1,
     }
+
+    learning_rate_kwargs = {
```
Comment (ericmjl) [inline jax_unirep/evotuning.py]:
```suggestion
    # Default settings for learning_rate_kwargs
    learning_rate_kwargs = {
```

Could you add an analogous comment on top of the `n_epochs_kwargs` line?

**Comment by ElArkk** [comment]
> Noice! Small changes requested for now.
> 
> One thing I didn't see, though, was stopping when NaNs are encountered. I only saw the assignment of loss to inf. Could you see where stopping on NaN could be added?

Didn't figure that out yet.

**Comment by ElArkk** [comment]
Just found this: https://optuna.readthedocs.io/en/latest/tutorial/pruning.html

Could be the thing we need to stop at inf loss.

**Comment by ericmjl** [comment]
No worries - I'm digging into the source as well. 

I think something along the lines of returning `state, loss` together and then checking `loss` in the outer loop to decide whether to break or not - that would be the way out.
